shopinvader_sale_profile/models/shopinvader_partner.py

```diff
@@ -0,0 +1,139 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, exceptions, fields, models, _
+from odoo.fields import first
+
+
+class ShopinvaderPartner(models.Model):
+    _inherit = 'shopinvader.partner'
+
+    sale_profile_id = fields.Many2one(
+        'shopinvader.sale.profile',
+        'Sale profile',
+        compute='_compute_sale_profile_id',
+        store=True,
+        help="Sale profile computed, depending every fields who make the "
+             "fiscal position (country, zip, vat, account position,...)",
+    )
+
+    @api.multi
+    @api.depends(
+        'record_id.country_id', 'country_id', 'record_id.vat', 'vat',
+        'record_id.property_product_pricelist', 'property_product_pricelist',
+        'property_account_position_id', 'state_id', 'zip',
+        'backend_id.use_sale_profile', 'backend_id.sale_profile_ids')
+    def _compute_sale_profile_id(self):
+        """
+        Compute function for the field sale_profile_id.
+        :return:
+        """
+        sale_profile_obj = self.env['shopinvader.sale.profile']
+        # Only backends using profiles
+        backend_ids = self.mapped("backend_id").filtered(
+            lambda b: b.use_sale_profile).with_prefetch(self._prefetch).ids
+        partners = self.mapped("record_id")
+        pricelists = partners.mapped("property_product_pricelist")
+        fposition_by_partner = self._get_fiscal_position_by_partner(partners)
+        # Get every fiscal position ids (without duplicates)
+        fposition_ids = list(set(fposition_by_partner.values()))
+        default_sale_profiles = self._get_default_profiles(backend_ids)
+        sale_profiles = self._get_sale_profiles(
+            backend_ids, pricelists, fposition_ids)
+        for binding in self:
+            sale_profile = sale_profile_obj.browse()
+            # Only if the related backend use sale profiles
+            if binding.backend_id.use_sale_profile:
+                partner = binding.record_id
+                fposition_id = fposition_by_partner.get(partner.id, False)
+                sale_profile = binding._sale_profile_with_backend(
+                    default_sale_profiles, fposition_id, sale_profiles)
+            binding.sale_profile_id = sale_profile
+
+    @api.multi
+    def _sale_profile_with_backend(self, default_sale_profiles,
+                                   fposition_id, sale_profiles):
+        """
+        Get the sale profile of current recordset based on default
+        sale profiles given in parameters, fiscal position id and
+        every related profile of related backend
+        :param default_sale_profiles: shopinvader.sale.profile recordset
+        :param fposition_id: int
+        :param sale_profiles: shopinvader.sale.profile recordset
+        :return: shopinvader.sale.profile recordset
+        """
+        self.ensure_one()
+        sale_profile = self.env['shopinvader.sale.profile'].browse()
+        partner = self.record_id
```

Comment:
@acsonefho @sebastienbeau Since ta company_id is specified on the backend, all the computation must be done with 'force_company' into the context (starting from `_compute_sale_profile_id`). Indeed some informations  are company dependent and the company context to take into account is not the user's company but the backend's company.